python/pyiceberg/table/__init__.py

```diff
@@ -365,8 +365,8 @@ class AssertDefaultSortOrderId(TableRequirement):
 
 class CommitTableRequest(IcebergBaseModel):
     identifier: Identifier = Field()
-    requirements: List[TableRequirement] = Field(default_factory=list)
-    updates: List[TableUpdate] = Field(default_factory=list)
+    requirements: List[SerializeAsAny[TableRequirement]] = Field(default_factory=list)
+    updates: List[SerializeAsAny[TableUpdate]] = Field(default_factory=list)
```

Comment:
Can you help me understand what's happening here?

Reply:
I also had to learn this by raising an issue: https://github.com/pydantic/pydantic/issues/6403#issuecomment-1620252270

It turns out that you need to wrap this when you're serializing an object, that might have subclasses. This way it gets correctly serialized.